src/irmin/metrics.mli

```diff
@@ -0,0 +1,80 @@
+(*
+ * Copyright (c) 2022 Etienne Marais <etienne@maiste.fr>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+(** The [Metrics] module defines primitives to handle metrics inside of Irmin.
+    Its purpose is to decouple the metrics definition from the data
+    manipulation. *)
+
+type origin = ..
+(** An extensible type to get the location of the definition. *)
+
+type origin +=
+  | Origin_default  (** Default {!origin} for the {!data} definition. *)
+
+type 'a metric = { name : string; uid : int; ty : 'a Repr.t }
+(** {!metric} caracterizes the type of the metric that is manipulated. *)
```

Comment:
Would be better to make it abstract - easier to extend in the future if needed.

Reply:
Yes, you are right!